fence/agents/scsi/fence_scsi.py

```diff
@@ -182,7 +182,14 @@ def get_cluster_id(options):
 	cmd = options["--corosync-cmap-path"] + " totem.cluster_name"
 
 	match = re.search(r"\(str\) = (\S+)\n", run_cmd(options, cmd)["out"])
-	return hashlib.md5(match.group(1)).hexdigest() if match else fail_usage("Failed: cannot get cluster name")
+
+	if not match:
+		fail_usage("Failed: cannot get cluster name")
+
+	try:
+		return hashlib.md5(match.group(1), usedforsecurity=False).hexdigest()
+	except TypeError:
+		return hashlib.md5(match.group(1)).hexdigest()
```

Comment:
Of course it will be, exactly at the point it is needed, without any
premature optimizations for a minor (as of now) use case.
There's hardly a better generic solution until Python upstream
figures the way forward -- note that there may be yet another
name for the keyword agument, or a completely different approach,
and FAs will be better off absorbing that approach as well if/when
it comes.